controlplane/pkg/nsm/nsm.go

```diff
@@ -46,32 +48,86 @@ func NewNetworkServiceManager(model model.Model, serviceRegistry serviceregistry
 	}
 }
 
-func (srv *networkServiceManager) Request(ctx context.Context, request nsm.NSMRequest, extra_parameters map[string]string) (nsm.NSMConnection, error) {
-	// Make sure its a valid request
+func (srv *networkServiceManager) Request(ctx context.Context, request nsm.NSMRequest) (nsm.NSMConnection, error) {
+	// Check if we are recovering connection, by checking passed connection Id is known to us.
+	return srv.request(ctx, request, srv.model.GetClientConnection(request.GetConnectionId()))
+}
+
+func create_logid() (uuid string) {
+	b := make([]byte, 4)
+	_, err := rand.Read(b)
+	if err != nil {
+		fmt.Println("Error: ", err)
+		return
+	}
+
+	uuid = fmt.Sprintf("%X", b[0:4])
+
+	return
+}
+
+func (srv *networkServiceManager) request(ctx context.Context, request nsm.NSMRequest, existingConnection *model.ClientConnection) (nsm.NSMConnection, error) {
+	requestId := create_logid()
+	logrus.Infof("NSM:(%v) request: %v", requestId, request )
+
+	// 0. Make sure its a valid request
 	err := request.IsValid()
 	if err != nil {
 		logrus.Error(err)
 		return nil, err
 	}
 
-	var nsmConnection nsm.NSMConnection = nil
-	nsmConnection = srv.cloneConnection(request, nsmConnection)
+	// 1. Create a new connection object.
+	nsmConnection := srv.newConnection(request)
 
-	// Create a ConnectId for the request.GetConnection(), since connections are managed per NSM
-	nsmConnection.SetId(srv.createConnectionId())
+	// 2. Set connection id for new connections.
```

Comment:
All these comments would make a great `docs/specs/AutoHealing.md`. Please consider it.

Reply:
Yes, planed to work on this as next task.